plugins/stages/list.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const joi = require('joi');
+const schema = require('screwdriver-data-schema');
+const jobIdSchema = schema.models.stage.base.extract('jobId');
+const listSchema = schema.models.stage.list;
+const nameSchema = schema.models.stage.base.extract('name');
+const pipelineIdSchema = schema.models.stage.base.extract('pipelineId');
+const jobIdsSchema = joi.alternatives().try(joi.array().items(jobIdSchema), jobIdSchema).required();
+const JOB_IDS_KEY = 'jobIds[]';
+
+module.exports = () => ({
+    method: 'GET',
+    path: '/stages',
+    options: {
+        description: 'Get stages',
+        notes: 'Returns all stage records',
+        tags: ['api', 'stages'],
+        auth: {
+            strategies: ['token'],
+            scope: ['user', 'build', 'pipeline']
+        },
+        handler: async (request, h) => {
+            const { stageFactory } = request.server.app;
+            const { name, pipelineId, sort, sortBy, page, count } = request.query;
+            const config = { sort };
+
+            // Only return specific stages
+            if (request.query[JOB_IDS_KEY]) {
+                const jobIds = request.query[JOB_IDS_KEY];
+
+                config.params.jobIds = Array.isArray(jobIds)
```

Comment:
This could throw error as `config.params` is `undefined`.
Applicable to other statements where `config.prams.` is being set.